Tests/ValetIntegrationTests/ValetIntegrationTests.swift

```diff
@@ -650,6 +677,22 @@ class ValetIntegrationTests: XCTestCase
         XCTAssertEqual(passcode, try vanillaValet.string(forKey: key))
     }
 
+    func test_removeObjectForKey_isDistinctForDifferingIdentifierInSameAccessGroup() throws
+    {
+        let valet1 = Valet.sharedGroupValet(with: Self.sharedAccessGroupIdentifier, identifier: Identifier(nonEmpty: "valet1")!, accessibility: vanillaValet.accessibility)
+        let valet2 = Valet.sharedGroupValet(with: Self.sharedAccessGroupIdentifier, identifier: Identifier(nonEmpty: "valet2")!, accessibility: vanillaValet.accessibility)
+
+        try valet1.setString(passcode, forKey: key)
+        try valet2.setString(passcode, forKey: key)
+
+        try valet2.removeObject(forKey: key)
+
+        XCTAssertEqual(passcode, try valet1.string(forKey: key))
+        XCTAssertThrowsError(try valet2.string(forKey: key)) { error in
+            XCTAssertEqual(error as? KeychainError, .itemNotFound)
+        }
+    }
+
```

Comment:
It might be nice to add a test that migrating between two different valets with different identifiers works as expected. (I know the test matrix is big + annoying but the more tests we have the more confident we are that this stuff actually works)

Reply:
The migration matrix doesn't seem to care about the *types* of the Valets that are involved in the migration, and more about the variations and keys and values. For example, there's nothing testing the migration of iCloud Valets to group access Valets to local Valets, etc. And I think that makes sense given that the migration uses the standard get/set/remove methods in all cases, and those are tested separately for each type.

Happy to add something here, but not entirely sure which tests of value are missing.

Reply:
Eh that's fair. I was going off memory. If we don't have similar tests then we're good to go here. I would, however, like to see test coverage improve (based on the code of comment), but I'm currently in mobile so I haven't checked details. Let me know if I can help offer possible tests to up coverage.

Reply:
The Codecov report says `The diff coverage is 100.00%.`, so I think all the new content is covered. It went down overall, but I'm not sure how that's calculated.